### ISSUE TITLE
define a definition.Body in consume Parameters, client don't pass contentType, should 400 not 500

### ISSUE THREAD

**Issue by zjj2wry** [caicloud/nirvana#200]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
/assign @kdada 

> /kind feature

**What happened**:
```
➜  licensed git:(9c81577) ✗ curl -X POST localhost:4242/api/licenses  --data '{"meta":{"name":"example","id":"12","createTimestamp":"0001-01-01T00:00:00Z"},"spec":{"company":{"name":"Company Name"},"licenseType":"POC","notBefore":"2018-03-27T18:04:23.054794897+08:00","notAfter":"2018-03-27T18:04:23.05479475+08:00","quota":{"cpu":"30"},"identify":"Cluster Identify"},"statuss":{"approve":false}}' -v
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 4242 (#0)
> POST /api/licenses HTTP/1.1
> Host: localhost:4242
> User-Agent: curl/7.58.0
> Accept: */*
> Content-Length: 318
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 318 out of 318 bytes
< HTTP/1.1 500 Internal Server Error
< Content-Type: application/json
< Date: Wed, 04 Apr 2018 07:57:05 GMT
< Content-Length: 225
< 
{"reason":"Nirvana:Service:InvalidTypeForConsumer","message":"consumer application/x-www-form-urlencoded can't consume data for type *api.License","data":{"content":"application/x-www-form-urlencoded","type":"*api.License"}}
* Connection #0 to host localhost left intact
```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Comment by zjj2wry** [comment]
@kdada @ddysher 
should we fix this problem? Because nirvana did some client doing things. When the client passed the wrong data, I think we shouldn't return 500

**Comment by kdada** [comment]
@zjj2wry Fix in v0.2.0
